tools/run_tests/xds_k8s_test_driver/framework/test_app/base_runner.py

```diff
@@ -35,14 +46,109 @@ class KubernetesBaseRunner:
     def __init__(self,
                  k8s_namespace,
                  namespace_template=None,
-                 reuse_namespace=False):
+                 reuse_namespace=False,
+                 case_name=None,
+                 pod_label_selector=None):
         # Kubernetes namespaced resources manager
         self.k8s_namespace: k8s.KubernetesNamespace = k8s_namespace
         self.reuse_namespace = reuse_namespace
         self.namespace_template = namespace_template or 'namespace.yaml'
+        self.case_name: str = case_name or ""
+        self.pod_label_selector: str = pod_label_selector or ""
 
         # Mutable state
         self.namespace: Optional[k8s.V1Namespace] = None
+        self._log_stop_event: threading.Event = threading.Event()
+
+    def _await_container_ready(self, core: client.CoreV1Api, pod_name: str,
+                               container_name: str, namespace: str) -> None:
+        while not self._log_stop_event.is_set():
+            try:
+                pod_status = core.read_namespaced_pod_status(
+                    pod_name, namespace).status
+                container = [
+                    container for container in pod_status.container_statuses
+                    if container.name == container_name
+                ][0]
```

Comment:
Do we need to worry about `IndexError`?

Reply:
This would indicate an error in the `kube-api-server`. But I suppose we should be resilient in that case too. Fixed.